### PR TITLE
feat(task): getStep, editStep, removeStep, cwd

### PR DESCRIPTION
**Pull request by dkershner6**
Fixes https://github.com/projen/projen/issues/1904

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mrgrain — COMMENTED** [review]
```
    // If the list of steps is a Lazy value, we can't know what the steps
    // are until synthesis occurs, so just return an empty array.
```

The `steps` getter has this note. How does this PR deal with this?

**Comment by dkershner6** [comment]
> ```
>     // If the list of steps is a Lazy value, we can't know what the steps
>     // are until synthesis occurs, so just return an empty array.
> ```
> 
> The `steps` getter has this note. How does this PR deal with this?

Added a warning for folks to try using the `preSynthesize` phase if they somehow run into that for all methods.  Good to look over once more, @mrgrain .

**Comment by dkershner6** [comment]
Tests failing on what seems like an errant type in a lib:

```
node_modules/@types/node/ts4.8/crypto.d.ts:3495:17 - error TS1110: Type expected.
```

File: src/task.ts [hunk 1]
```diff
@@ -154,6 +154,14 @@ export class Task {
    */
   public exec(command: string, options: TaskStepOptions = {}) {
     this.assertUnlocked();
+
+    if (!Array.isArray(this._steps)) {
+      warn(
+        `Cannot add exec to task "${this.name}" because it is a lazy value, try using the preSynthesize phase.`
+      );
+      return;
+    }
```
Comment (mrgrain) [inline src/task.ts:163]: Nice and thanks for adding this everywhere! Can you pull this out into a helper, similar to `assertUnlocked()` so we don't have to repeat ourselves.

**Review by mrgrain — CHANGES_REQUESTED** [review]

**Comment by dkershner6** [comment]
@mrgrain , I realized that `reset()` would also have failed, so I added a check there as well, but all now use a centralized helper.  Feel free to have another look at your convenience.

**Review by mrgrain — APPROVED** [review]

**Pull request merged**